### ISSUE TITLE
Can't build for mantic (23.10) even on latest/edge

### ISSUE THREAD

**Issue by ajkavanagh** [canonical/charmcraft#1203]
### Bug Description

I would like to build charms for mantic/23.10.  Unfortunately, this is the result (even with latest/edge):

```
charmcraft -v pack --bases-index=8
Starting charmcraft version 2.4.0.post2+git31f7281
Logging execution to '/home/alex/.local/state/charmcraft/log/charmcraft-20230807-114746.906958.log'
Packing the charm.
Skipping 'bases[8].build-on[0]': base 'ubuntu' channel '23.10' is not yet supported (must be 'almalinux 9', 'centos 7', 'ubuntu 16.04', 'ubuntu 18.04', 'ubuntu 20.04', 'ubuntu 22.04',
 'ubuntu 22.10', 'ubuntu 23.04' or 'ubuntu devel').
```

### To Reproduce

Use a charmcraft with bases containing:

```yaml
  - build-on:
      - name: ubuntu
        channel: "23.10"
        architectures: [amd64]
    run-on:
      - name: ubuntu
        channel: "23.10"
        architectures: [amd64]
```

### Environment

Ubuntu 22.04 LTS.

### charmcraft.yaml

```shell
type: charm

parts:
  charm:
    plugin: reactive
    reactive-charm-build-arguments:
      - --binary-wheels-from-source
    build-packages:
      - tox
      - git
      - python3-dev
      - libffi-dev
      - libpython3-dev
      - libssl-dev
      - libxml2-dev
      - libxslt1-dev
      - rustc
      - cargo
      - pkg-config
    source: src/
    build-snaps:
      - charm/latest/edge
    build-environment:
      - CHARM_INTERFACES_DIR: $CRAFT_PROJECT_DIR/interfaces/
      - CHARM_LAYERS_DIR: $CRAFT_PROJECT_DIR/layers/
bases:
  - build-on:
      - name: ubuntu
        channel: "23.10"
        architectures: [amd64]
    run-on:
      - name: ubuntu
        channel: "23.10"
        architectures: [amd64]
```

### Relevant log output

```shell
$ charmcraft -v pack --bases-index=8
Starting charmcraft version 2.4.0.post2+git31f7281
Logging execution to '/home/alex/.local/state/charmcraft/log/charmcraft-20230807-115143.616635.log'
Packing the charm.
Skipping 'bases[8].build-on[0]': base 'ubuntu' channel '23.10' is not yet supported (must be 'almalinux 9', 'centos 7', 'ubuntu 16.04', 'ubuntu 18.04', 'ubuntu 20.04', 'ubuntu 22.04', 'ubuntu 22.10', 'ubuntu 23.04' or 'ubuntu devel').
No suitable 'build-on' environment found in 'bases[8]' configuration.
No suitable 'build-on' environment found in any 'bases' configuration.
Full execution log: '/home/alex/.local/state/charmcraft/log/charmcraft-20230807-115143.616635.log'
```

**Comment by syu-w** [comment]
23.10 is not supported at the moment. It was a buildd image issue, and I haven't check if it fixed or not.

https://github.com/canonical/craft-providers/pull/330

**Comment by lengau** [comment]
@ajkavanagh this should be fixed on edge today and in charmcraft 2.5 coming later today.

**Comment by ajkavanagh** [comment]
> @ajkavanagh this should be fixed on edge today and in charmcraft 2.5 coming later today.

Thanks very much!  Much appreciated.
